src/v/cloud_storage/remote_partition.cc

```diff
@@ -136,13 +136,17 @@ remote_partition::borrow_result_t remote_partition::borrow_next_segment_reader(
         }
         if (mit == manifest.end()) {
             // Segment that matches exactly can't be found in the manifest. In
-            // this case we want to start scanning from the begining of the
+            // this case we want to start scanning from the beginning of the
             // partition if the start of the manifest is contained by the scan
             // range.
+            // Opt out in case of timequery. Timequery scans from the beginning
+            // if query undershoots the data in the manifest and it should not
+            // re-scan the partition in case if query overshoots.
             auto so = manifest.get_start_kafka_offset().value_or(
               kafka::offset::min());
             if (
-              model::offset_cast(config.start_offset) < so
+              config.first_timestamp.has_value() == false
```

Comment:
what is this part of the conjunction?
> config.first_timestamp.has_value() == false

is that referring to this from the commit message?

> didn't find the segment